### PR TITLE
Fix rust-gdb on FreeBSD

### LINKED ISSUE #110334: rust-gdb crashes on FreeBSD due to invalid sed syntax
### Code

I tried this code:

```sh
rust-gdb
```

I expected to see this happen: GDB starts

Instead, this happened: GDB crashed:
```
$ rust-gdb
sed: 1: "s/commit-hash: \(\w*\)/ ...": RE error: trailing backslash (\)
```

### Version it worked on

It most recently worked on:
rustc 1.70.0-nightly (4a04d086c 2023-03-18)

### Version with regression
```
rustc 1.70.0-nightly (a41fc00ea 2023-04-13)
binary: rustc
commit-hash: a41fc00eaf352541008965fec0dee811e44373b3
commit-date: 2023-04-13
host: x86_64-unknown-freebsd
release: 1.70.0-nightly
LLVM version: 16.0.2
```
The bug was introduced by commit 8dd0ec6de7c8473e5730972c4e6206d5fb47b4e6 , which invokes sed with GNU-specific syntax.

@rustbot modify labels: +regression-from-stable-to-nightly -regression-untriaged

### PR DESCRIPTION
**Pull request by asomers**
"\w" is a GNU-specific extension to sed.  Avoid it.

Fixes #110334
Signed-off-by: Alan Somers <asomers@gmail.com>

### CODE REVIEW

**Comment by cuviper** [comment]
`rust-gdbgui` has the same regex.

**Comment by asomers** [comment]
Good catch.  I force-pushed to fix it on rust-gdbgui too.

File: src/etc/rust-gdb [hunk 1]
```diff
@@ -14,7 +14,7 @@ fi
 RUSTC_SYSROOT="$("$RUSTC" --print=sysroot)"
 GDB_PYTHON_MODULE_DIRECTORY="$RUSTC_SYSROOT/lib/rustlib/etc"
 # Get the commit hash for path remapping
-RUSTC_COMMIT_HASH="$("$RUSTC" -vV | sed -n 's/commit-hash: \(\w*\)/\1/p')"
+RUSTC_COMMIT_HASH="$("$RUSTC" -vV | sed -n 's/commit-hash: \([a-zA-z0-9_]*\)/\1/p')"
```
Comment (SNCPlay42) [inline src/etc/rust-gdb:17]:
```suggestion
RUSTC_COMMIT_HASH="$("$RUSTC" -vV | sed -n 's/commit-hash: \([a-zA-Z0-9_]*\)/\1/p')"
```

Comment (cuviper) [reply src/etc/rust-gdb:17]: Nice catch!
@bors r-

**Pull request merged**